### PR TITLE
FindSimilarIncidents - improve query performance

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28537

## Description
Populate `fromdate` and `todate` arguments when execute `getIncidents` command instead of using the query text

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/ReleaseNotes/1_3_41.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+#### Scripts
+##### FindSimilarIncidents
+- Improve query performance when using **hoursBack** flag.
+- Updated the Docker image to: *demisto/python:2.7.18.18413*.
```
Comment (yaakovi) [inline Packs/CommonScripts/ReleaseNotes/1_3_41.md:5]: you didn't update the yml with the new docker

File: Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py [hunk 2]
```diff
@@ -138,27 +136,26 @@ def get_incidents_by_keys(similar_incident_keys, time_field, incident_time, inci
     incident_time = parse_datetime(incident_time)
     max_date = incident_time
     min_date = incident_time - timedelta(hours=hours_back)
-    hours_back_query = '{0}:>="{1}" and {0}:<"{2}"'.format(time_field, min_date.isoformat(), max_date.isoformat())
+    query = ''
```
Comment (yaakovi) [inline Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py:139]: add unit tests for the query construction if there are none

Comment (adi88d) [reply Packs/CommonScripts/Scripts/FindSimilarIncidentsV2/FindSimilarIncidentsV2.py]: I split the query construction part to different function with the name `build_incident_query` and added unit tests for this function.

**Review by yaakovi — APPROVED** [review]
